Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/extractors/BaseEmailExtractor.java

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.recognizers.text.sequence.extractors;
+
+import com.microsoft.recognizers.text.ExtractResult;
+import com.microsoft.recognizers.text.sequence.Constants;
+import com.microsoft.recognizers.text.sequence.SequenceOptions;
+import com.microsoft.recognizers.text.sequence.config.BaseSequenceConfiguration;
+import com.microsoft.recognizers.text.sequence.resources.BaseEmail;
+import com.microsoft.recognizers.text.utilities.StringUtility;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class BaseEmailExtractor extends BaseSequenceExtractor {
+    private static final Pattern RFC_5322_VALIDATION_REGEX = Pattern.compile(BaseEmail.RFC5322Regex);
+    private final BaseSequenceConfiguration config;
+
+    protected final String extractType = Constants.SYS_EMAIL;
+
+    protected Map<Pattern, String> getRegexes() {
+        return regexes;
+    }
+
+    protected String getExtractType() {
+        return extractType;
+    }
+
+    @Override
+    protected List<ExtractResult> postFilter(List<ExtractResult> results) {
+        // If Relaxed is on, no extra validation is applied
+        if (config.getOptions() != SequenceOptions.None) {
+            return results;
+        } else {
+            // Not return malformed e-mail addresses and trim ending '.'
+            results.forEach(result -> {
+                if (result.getText().endsWith(".")) {
+                    result.setText(StringUtility.trimEnd(result.getText()));
+                    result.setLength(result.getLength() - 1);
+                }
+            });
+        }
+
+        return results.stream().filter((o -> RFC_5322_VALIDATION_REGEX.matcher((o).getText()).matches()))
+                .collect(Collectors.toList());
+    }
+
+    public BaseEmailExtractor(BaseSequenceConfiguration config) {
+        this.config = config;
+        Map<Pattern, String> regexes = new HashMap<Pattern, String>();
+        regexes.put(Pattern.compile(BaseEmail.EmailRegex), Constants.EMAIL_REGEX);
+        // EmailRegex2 will break the code as it's not supported in Java, comment out for now
+        // regexes.put(Pattern.compile(BaseEmail.EmailRegex2), Constants.EMAIL_REGEX);
```

Comment:
Why is it not supported in Java? What are the side-effects of commenting this out?
In such cases, please add a `@TODO` comment to implement a solution or workaround.

Reply:
We [added](https://github.com/microsoft/Recognizers-Text/pull/2465/commits/78e2353a0814d0f2c345079381805d899ae2368b) a `@TODO` comment. The approach was the same as JavaScript which was added in PR #905 and also like Python in PR #1489 (see [comment](https://github.com/microsoft/Recognizers-Text/pull/1489#discussion_r264083539))

_Exception is EmailRegex2 is commented out_
![image](https://user-images.githubusercontent.com/11904023/107406876-b618cb80-6ae7-11eb-830b-59cba552233a.png)